instaloader/instaloader.py

```diff
@@ -204,6 +206,26 @@ def __init__(self,
         self.resume_prefix = resume_prefix
         self.check_resume_bbd = check_resume_bbd
 
+        self.slide = slide or ""
+        self.slide_start = 0
+        self.slide_end = 0
+        if self.slide != "":
+            splitted = self.slide.split('-')
+            try:
+                if len(splitted) == 1:
+                    self.slide_start = self.slide_end = int(splitted[0])
+                    if self.slide_start < 0:
+                        raise InvalidArgumentException("Invalid data for --slide parameter")
+                else:
+                    if int(splitted[0]) < int(splitted[1]) and int(splitted[0]) > 0:
+                        self.slide_start = int(splitted[0])
+                        self.slide_end = int(splitted[1])
+                    else:
+                        raise InvalidArgumentException("Invalid data for --slide parameter")
```

Comment:
We could easily catch the case of `len(splitted) > 2` here, by nesting the if/else as follows:

```suggestion
                elif len(splitted) == 2 and int(splitted[0]) < int(splitted[1]) and int(splitted[0]) > 0:
                    self.slide_start = int(splitted[0])
                    self.slide_end = int(splitted[1])
                else:
                    raise InvalidArgumentException("Invalid data for --slide parameter: ".format(slide))
```